#! /bin/sh -e
## 36_sb1-addrspace.dpatch by "Mark Mason" <mason@broadcom.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Missing addrspace defines for sb-1.

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p0 ${patch_opts} < $0;;
    -unpatch) patch -R -p0 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

Here's the patch for that particular problem.  There's also a few other 
patches for the SB1 floating around (check the email archives), but 
there appears to be a backlog with getting them committed to the CVS 
repository.

Index: include/asm-mips/addrspace.h
===================================================================
RCS file: /home/cvs/linux/include/asm-mips/addrspace.h,v
retrieving revision 1.18
diff -u -p -r1.18 addrspace.h
--- include/asm-mips/addrspace.h    14 Jul 2005 12:05:08 -0000    1.18
+++ include/asm-mips/addrspace.h    15 Sep 2005 21:46:56 -0000
@@ -131,6 +131,8 @@
     || defined (CONFIG_CPU_R5000)					\
     || defined (CONFIG_CPU_NEVADA)					\
     || defined (CONFIG_CPU_TX49XX)					\
+    || defined (CONFIG_CPU_SB1)						\
+    || defined (CONFIG_CPU_SB1A)					\
     || defined (CONFIG_CPU_MIPS64)
 #define KUSIZE		_LLCONST_(0x0000010000000000)	/* 2^^40 */
 #define KUSIZE_64	_LLCONST_(0x0000010000000000)	/* 2^^40 */
